python/cuml/cuml/model_selection/_split.py

```diff
@@ -363,55 +373,63 @@ def train_test_split(
         train_indices = range(0, train_size)
         test_indices = range(-1 * test_size, 0)
 
-    # Gather from indices
-    X_train = X_arr[train_indices]
-    X_test = X_arr[test_indices]
-    if y is not None:
-        y_train = y_arr[train_indices]
-        y_test = y_arr[test_indices]
+    if all_numeric:
+        # Gather from indices
+        X_train = X_arr[train_indices]
+        X_test = X_arr[test_indices]
+        if y is not None:
+            y_train = y_arr[train_indices]
+            y_test = y_arr[test_indices]
+
+        # Coerce output to original input type
+        if ty := determine_df_obj_type(X):
```

Comment:
Can this be simplified to `x_type = determine_df_obj_type(X) or determine_array_type(X)`? Similar for y and other places where this pattern occurs.